### ISSUE TITLE
Unable to define repo url in Docker config via env

### ISSUE THREAD

**Issue by tebruno99** [goreleaser/goreleaser#2281]
Attempting to use Env vars in dockers: config section
```
dockers:
  -
    goos: linux
    goarch: amd64
    ids:
      - demourl
    skip_push: false
    dockerfile: Dockerfile
    image_templates:
      - "{{ .Env.AWS_ECR_REPO }}:latest"
      - "{{ .Env.AWS_ECR_REPO }}:{{ .Tag }}"

```
Command:
```AWS_ECR_REPO="demorepo.dkr.ecr.us-west-2.amazonaws.com/demourl" goreleaser```

This results in the error:
```      • building docker image     buildx=false image={{ .Env.AWS_ECR_REPO }}:latest
   ⨯ release failed after 3.05s error=failed to build docker image: {{ .Env.AWS_ECR_REPO }}:latest: 
invalid argument "{{ .Env.AWS_ECR_REPO }}:latest" for "-t, --tag" flag: invalid reference format: repository name must be lowercase
```

Rather than hard coding a repo I wanted to pass in the url from Travis/Github Actions so when someone forks they can define their own. I expected the value of .Env.AWS_ECR_REPO to be used during the docker build instead of the template text itself.

**Environment (please complete the following information):**

- OS: linux
- OS version: Ubuntu 21.04
- GoReleaser Version: 0.168.0

**Comment by caarlos0** [comment]
That should actually work (just added more tests enforcing it)... can't see how the template could not be parsed 🤔  if the env was not present for some reason, it would have errored with a `map has no entry for key` error.

can you provide a reproducible?

**Comment by tebruno99** [comment]
I setup an example here:  https://github.com/tebruno99/hello

I had copied the example on the documentation page "Environment Variables" to provide a default local docker repo if an ECR repo was not provided. 
```
env:
  - GO111MODULE=on
  - AWS_ECR_REPO={{ if index .Env "AWS_ECR_REPO" }}{{ .Env.AWS_ECR_REPO }}{{ else }}tebruno99{{ end }}
```

I've tried this with both the OSS and Pro version and neither worked. I may be misunderstanding something I'm not super familiar with go templates.
